### DISCUSSION TITLE
bot.launch() not working as expected

### DISCUSSION THREAD

**Discussion by ymittal** [telegraf/telegraf#1884] [Q&A]
Hey, I was expecting `bot.launch` to always finish but somehow it's not working. I am not sure what I am doing wrong.
```
(async () => {
  try {
    await bot.launch({});
    console.log("Telegram bot is ready! 🤖");
  } catch (error) {
    console.error("Error starting Telegram bot:", error);
  }
})();
```

I have also tried `bot.launch().then(() => { ... })` but that also doesn't work.

---
Btw, I have a sync job which runs every minute. I want to restrict/unrestrict users in chats that my bot has been added to. What's the best way to maintain a global `ctx` object so I can use `ctx.telegram.restrictChatMember`

**Comment by MKRhere** [comment]
1. See https://github.com/telegraf/telegraf/issues/1749

2. Use `bot.telegram.restrictChatMember`.
